nvtabular/ops/data_stats.py

```diff
@@ -13,95 +13,290 @@
+
+        if not os.path.exists(stats_file):
+            return None
+
+        LOG.info("Loading stats from %s", stats_file)
+        d = statistics_pb2.DatasetFeatureStatisticsList()
+        with open(stats_file, "rb") as f:
+            d.ParseFromString(f.read())
+
+        return DatasetCollectionStatistics(d)
 
     def transform(self, columns: Columns, df: DataFrameType) -> DataFrameType:
         return df
 
-    @annotate("DataStats_fit", color="green", domain="nvt_python")
+    @annotate("Statistics_fit", color="green", domain="nvt_python")
     def fit(self, columns: Columns, ddf: dd.DataFrame):
         dask_stats = {}
 
         ddf_dtypes = ddf.head(1)
+        num_examples = len(ddf)
+
+        #         dask_stats["median"] = ddf[columns].quantile(q=0.5, method="dask")
 
         # For each column, calculate the stats
-        for col in columns.names():
+        for col in columns:
             dask_stats[col] = {}
             self.col_names.append(col)
             # Get dtype for all
-            dtype = ddf_dtypes[col].dtype
+            dtype = str(ddf_dtypes[col].dtype)
+            is_list = False
+
+            domain = ddf[col]
+
+            if dtype == "list":
+                domain = ddf[col].map_partitions(lambda x: x.list.leaves, meta=("x", int))
+                lengths = ddf[col].map_partitions(lambda x: x.list.len(), meta=("x", int))
+                dask_stats[col]["min_length"] = lengths.min()
+                dask_stats[col]["max_length"] = lengths.max()
+                dask_stats[col]["mean_length"] = lengths.mean()
+                dtype = ddf_dtypes[col].dtype.leaf_type
+                is_list = True
+
             self.col_dtypes.append(dtype)
+            self.col_is_lists.append(is_list)
 
-            # Identify column type
-            if np.issubdtype(dtype, np.floating):
-                col_type = "conts"
-            else:
-                col_type = "cats"
-            self.col_types.append(col_type)
+            dask_stats[col]["num_missing"] = len(domain[domain.isnull()])
+            dask_stats[col]["num_non_missing"] = domain.count()
 
             # Get cardinality for cats
-            if col_type == "cats":
-                dask_stats[col]["cardinality"] = ddf[col].nunique()
-
-            # if string, replace string for their lengths for the rest of the computations
             if dtype == "object":
-                ddf[col] = ddf[col].map_partitions(lambda x: x.str.len(), meta=("x", int))
-            # Add list support when cudf supports it:
-            # https://github.com/rapidsai/cudf/issues/7157
-            # elif col_type == "cat_mh":
-            #    ddf[col] = ddf[col].map_partitions(lambda x: x.list.len())
+                dask_stats[col]["nunique"] = domain.nunique()
+                str_len = domain.map_partitions(lambda x: x.str.len(), meta=("x", int))
+                dask_stats[col]["avg_str_length"] = str_len.mean()
```

Comment:
Can you get the min and max string lengths also... this is helpful for transferring to the data generator.